src/main/java/com/hubspot/jinjava/el/ext/eager/EagerExtendedParser.java

```diff
@@ -58,6 +64,21 @@ public EagerExtendedParser(Builder context, String input) {
     );
   }
 
+  @Override
+  protected AstEval eval(boolean required, boolean deferred)
+    throws ScanException, ParseException {
+    AstEval v = null;
+    Symbol startEval = deferred ? START_EVAL_DEFERRED : START_EVAL_DYNAMIC;
+    if (getToken().getSymbol() == startEval) {
+      consumeToken();
+      v = new AstEval(new EagerAstRoot(expr(true)), deferred);
```

Comment:
This is added so that we have a way to just call `clearEvalResult()` after all the AST nodes have been processed, rather than as a `finally` clause at the end of each node.

(By the way, we need to clear the evaluation result because a Cache is used for the AstNodes, if they have the same string image as one that was used in a previous ELExpression, then it will be reused so we need to make sure that all the eval results for each Eager ast node is cleared after the current EL expression is evaluated.